Q3DC/Q3DC.py

```diff
@@ -369,6 +392,105 @@ def UpdateInterface(self):
                                                     self.fidListComboBoxlineLB.currentNode())
         self.logic.UpdateThreeDView(self.landmarkComboBox.currentText)
 
+    def init_anatomical_legend(self):
+        if self.anatomical_legend is None:
+            for table_node in slicer.mrmlScene.GetNodesByClass('vtkMRMLTableNode'):
+                if table_node.GetAttribute('Q3DC.is_anatomical_legend') == 'True':
+                    self.anatomical_legend = table_node
+            if self.anatomical_legend is None:
+                self.anatomical_legend = slicer.vtkMRMLTableNode()
+                self.anatomical_legend.SetSaveWithScene(False)
+                self.anatomical_Legend.SetLocked(True)
+                slicer.mrmlScene.AddNode(self.anatomical_legend)
+                self.anatomical_legend.SetAttribute('Q3DC.is_anatomical_legend', 'True')
+
+        al = self.anatomical_legend
+        modify = al.StartModify()
```

Comment:
Consider using `NodeModify` context manager. See [here](https://github.com/Slicer/Slicer/blob/6256f99080957f39f41b1830f52bd8dcd5263a3d/Modules/Scripted/VectorToScalarVolume/VectorToScalarVolume.py#L266) for an example